graph/api/src/main/java/org/jboss/windup/graph/service/GraphService.java

```diff
@@ -224,20 +247,37 @@ public T getUnique() throws NonUniqueResultException
     @Override
     public T getUniqueByProperty(String property, Object value) throws NonUniqueResultException
     {
-        Iterable<T> results = findAllByProperty(property, value);
+        return getUniqueByProperty(property, value, false);
+    }
 
-        if (!results.iterator().hasNext())
-        {
-            return null;
-        }
+    public T getUniqueByProperty(String property, Object value, boolean enforceType) throws NonUniqueResultException
+    {
+        Iterable<T> results = findAllByProperty(property, value, enforceType);
 
-        Iterator<T> iterator = results.iterator();
-        T result = iterator.next();
+        T result = null;
+        Iterator<? extends WindupVertexFrame> iterator = results.iterator();
+        do {
+            if (!iterator.hasNext())
+                return result;
 
-        if (iterator.hasNext())
-        {
-            throw new NonUniqueResultException("Expected unique value, but returned non-unique.");
+            WindupVertexFrame item = iterator.next();
+
+            // There can be other types using the same property name.
+            if (!type.isInstance(item))
+                continue;
+
+            if ("jsf".equals(value))            /// DEBUG, remove
```

Comment:
I got this entry in the log file:
`[FOO] (main) Model with name 'jsf', type: interface org.jboss.windup.reporting.model.TagModel item: interface org.jboss.windup.graph.model.WindupVertexFrame, interface com.tinkerpop.frames.VertexFrame, interface org.jboss.windup.reporting.model.TagModel`
Is it useful to debug? Or can it be removed?

Reply:
lol, how did I not notice that?